src/core/lib/transport/metadata.cc

```diff
@@ -673,6 +673,10 @@ void grpc_mdelem_do_unref(grpc_mdelem gmd DEBUG_ARGS) {
 
 void grpc_mdelem_on_final_unref(grpc_mdelem_data_storage storage, void* ptr,
                                 uint32_t hash DEBUG_ARGS) {
+#ifndef NDEBUG
+  (void)file;
+  (void)line;
+#endif
```

Comment:
Why is this needed here? This function doesn't reference file or line variables anywhere.

Reply:
This is needed because of `DEBUG_ARGS` which defines file and line. `DEBUG_ARGS` seems possible to remove for this function but i'm not 100% sure.